tests/components/netatmo/test_sensor.py

```diff
@@ -0,0 +1,233 @@
+"""The tests for the Netatmo sensor platform."""
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components.netatmo import sensor
+from homeassistant.components.netatmo.helper import NetatmoArea
+from homeassistant.components.netatmo.sensor import MODULE_TYPE_WIND
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .common import TEST_TIME
+from .conftest import selected_platforms
+
+
+async def test_weather_sensor(hass, sensor_entry):
+    """Test weather sensor setup."""
+    prefix = "sensor.netatmo_mystation_"
+
+    assert hass.states.get(f"{prefix}temperature").state == "24.6"
+    assert hass.states.get(f"{prefix}humidity").state == "36"
+    assert hass.states.get(f"{prefix}co2").state == "749"
+    assert hass.states.get(f"{prefix}pressure").state == "1017.3"
+
+
+async def test_public_weather_sensor(hass, sensor_entry):
+    """Test public weather sensor setup."""
+    await hass.async_block_till_done()
+
+    prefix = "sensor.netatmo_home_max_"
+
+    assert hass.states.get(f"{prefix}temperature").state == "27.4"
+    assert hass.states.get(f"{prefix}humidity").state == "76"
+    assert hass.states.get(f"{prefix}pressure").state == "1014.4"
+
+    prefix = "sensor.netatmo_home_avg_"
+
+    assert hass.states.get(f"{prefix}temperature").state == "22.7"
+    assert hass.states.get(f"{prefix}humidity").state == "63.2"
+    assert hass.states.get(f"{prefix}pressure").state == "1010.3"
+
+    # Test adding new public area
+    area_a = NetatmoArea(
+        lat_ne=32.2345678,
+        lon_ne=-117.1234567,
+        lat_sw=32.1234567,
+        lon_sw=-117.2345678,
+        show_on_map=False,
+        area_name="Home avg",
+        mode="avg",
+    )
+    async_dispatcher_send(
```

Comment:
Don't use the dispatch helper directly. What is triggering the dispatch? Is it a webhook? Then use the webhook handler from the webhook integration as trigger.

Reply:
This is triggered from the config options.